tfjs-backend-webgpu/src/kernels/crop_and_resize_webgpu.ts

```diff
@@ -25,29 +25,26 @@ export class CropAndResizeProgram implements WebGPUProgram {
   dispatchLayout: {x: number[]};
   dispatch: [number, number, number];
   variableNames = ['Image', 'Boxes', 'BoxInd'];
+  uniforms = 'float extrapolationValue;';
   workGroupSize: [number, number, number] = [64, 1, 1];
   imageShape: [number, number, number, number];
   cropSize: [number, number];
   methodId: number;
-  extrapolationValue: number;
 
   constructor(
       imageShape: [number, number, number, number], boxShape: [number, number],
-      cropSize: [number, number], method: 'bilinear'|'nearest',
-      extrapolationValue: number) {
+      cropSize: [number, number], method: 'bilinear'|'nearest') {
     const [numBoxes, ] = boxShape;
     this.outputShape = [numBoxes, cropSize[0], cropSize[1], imageShape[3]];
     this.dispatchLayout = flatDispatchLayout(this.outputShape);
     this.dispatch = computeDispatch(
         this.dispatchLayout, this.outputShape, this.workGroupSize);
 
-    // imageShape is used by const imageHeight and imageWidth.
-    this.shaderKey = `cropAndResize_${method}_${cropSize}_${
-        extrapolationValue}_${imageShape}`;
     this.imageShape = imageShape;
     this.cropSize = cropSize;
     this.methodId = method === 'bilinear' ? 1 : 0;
-    this.extrapolationValue = extrapolationValue;
+    // imageShape is used by const imageHeight and imageWidth.
+    this.shaderKey = `cropAndResize_${this.methodId}_${cropSize}_${imageShape}`;
```

Comment:
Please also remove the dependency for `cropSize` and `imageShape`. I think `cropSize` is equals to `ivec2(outShape[1], outShape[2])`. And outShape and imageShape are already uniforms. You can directly use them.

Reply:
done